.github/workflows/1-create-a-pr.yml

```diff
@@ -1,93 +1,62 @@
-name: Step 1, Create a pull request
+name: Step 1
 
-# This step triggers after the learner creates a pull request for the my-resume branch
-# This workflow updates from step 1 to step 2.
-
-# This will run every time we open a pull request.
-# Reference: https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows
 on:
-  workflow_dispatch:
-  pull_request:
+  pull_request_target:
+    branches: [main]
     types: [opened]
+  workflow_dispatch: {}
 
-# Reference: https://docs.github.com/en/actions/security-guides/automatic-token-authentication
 permissions:
-  # Need `contents: read` to checkout the repository.
-  # Need `contents: write` to update the step metadata.
   contents: write
+  actions: write
+  issues: write
+  pull-requests: write  # Required for commenting or updating PR
 
-jobs:
-  # Get the current step to only run the main job when the learner is on the same step.
-  get_current_step:
-    name: Check current step number
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v4
-      - id: get_step
-        run: |
-          echo "current_step=$(cat ./.github/steps/-step.txt)" >> $GITHUB_OUTPUT
-    outputs:
-      current_step: ${{ steps.get_step.outputs.current_step }}
-
-  on_create_pr:
-    name: On create PR
-    needs: get_current_step
+env:
+  STEP_2_FILE: ".github/steps/2-resolve-a-merge-conflict.md"
 
-    # We will only run this action when:
-    # 1. This repository isn't the template repository.
-    # 2. The step is currently 1.
-    # Reference: https://docs.github.com/en/actions/learn-github-actions/contexts
-    # Reference: https://docs.github.com/en/actions/learn-github-actions/expressions
-    if: >-
-      ${{ !github.event.repository.is_template
-          && needs.get_current_step.outputs.current_step == 1 }}
+jobs:
+  find_exercise:
+    name: Find Exercise Issue
+    uses: skills/exercise-toolkit/.github/workflows/find-exercise-issue.yml@v0.5.0
 
-    # We'll run Ubuntu for performance instead of Mac or Windows.
+  post_next_step_content:
+    name: Post next step content
+    needs: [find_exercise]
     runs-on: ubuntu-latest
+    env:
+      ISSUE_URL: ${{ needs.find_exercise.outputs.issue-url }}
 
     steps:
-      # We'll need to check out the repository so that we can edit the README.
-      - name: Checkout
+      - name: Checkout the base branch (main)
         uses: actions/checkout@v4
         with:
-          fetch-depth: 0 # Let's get all the branches.
-
-      - name: Prepare a merge conflict
-        run: |
-          echo "Make sure we are on step 1"
-          if [ "$(cat .github/steps/-step.txt)" != 1 ]
-          then
-            echo "Current step is not 1"
-            exit 0
-          fi
+          ref: main  # Required because pull_request_target runs in the base branch context
```

Comment:
This wouldn't be needed on `pull_request` event trigger. Is there any reason for using this + `pull_request_target` ?